source/puppet/4.8/env_isolation.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: "Isolating resources in multiple environments"
+canonical: "/puppet/latest/environment_isolation.html"
+description: "Generating metadata to isolate resources in environments in Puppet"
+---
+
+[code_mgr_env]: {{pe}}/code_mgr.html#environment-isolation-metadata
+
+
+If you have multiple environments, environment isolation prevents resource types from leaking between your various environments. To use environment isolation, you'll generate metadata files that Puppet can use in place of the normal Ruby resource type implementations.
```

Comment:
@KevinCorcoran and @hlindberg : In the short description at the top (the graf Kevin quoted), I can say "module resource types," if that is accurate. (I'm not sure how descriptive "custom resource types" is for less experienced users.) In the longer description of the problem, below that, I can include a statement: "This problem occurs with custom resource types in modules, not with Puppet's built-in resource types."

Does that seem accurate and reasonable? I don't know if custom types are used only within modules...?